src/test/java/com/epam/tc/hw4/steps/HomePageSteps.java

```diff
@@ -0,0 +1,96 @@
+package com.epam.tc.hw4.steps;
+
+import static com.epam.tc.hw3.tests.utils.DataClass.EXPECTED_HEADER_NAV_BAR_ITEM_TEXTS;
+import static com.epam.tc.hw3.tests.utils.DataClass.EXPECTED_SIDE_NAV_BAR_ITEM_TEXTS;
+import static com.epam.tc.hw3.tests.utils.DataClass.EXPECTED_TEXTS_UNDER_THE_IMAGES;
+
+import com.epam.tc.hw4.pages.HomePage;
+import com.epam.tc.hw4.tests.utils.BaseTest;
+import com.epam.tc.hw4.tests.utils.TestProperties;
+import io.qameta.allure.Step;
+import org.openqa.selenium.WebDriver;
+import org.testng.asserts.SoftAssert;
+
+public class HomePageSteps {
+    protected SoftAssert softAssert;
+    protected HomePage homePage;
+    protected BaseTest baseTest;
+
+    public HomePageSteps(WebDriver webDriver) {
+        homePage = new HomePage(webDriver);
+        softAssert = new SoftAssert();
+    }
+
+    @Step("Open Home Page")
+    public void openHomePage() {
+        homePage.openHomePage();
+    }
+
+    @Step("Assert Home Page title")
+    public void assertHomePageTitle() {
+        softAssert.assertTrue(homePage.getDriver().getTitle().contains("Home Page"));
+    }
+
+    @Step("Assert incorrect Home Page title")
+    public void assertIncorrectHomePageTitle() {
+        softAssert.assertTrue(homePage.getDriver().getTitle().contains("Test Page"));
+    }
+
+    @Step("Perform Login")
+    public void performLogin() {
+        homePage.getHeader().login();
+    }
+
+    @Step("Assert Username")
+    public void assertUsername() {
+        softAssert.assertEquals(homePage.getHeader().getUserName(),
+            TestProperties.getTestProperties().getProperty("fullName"));
+    }
+
+    @Step("Assert header navbar items")
+    public void assertHeaderNavBarItems() {
+        softAssert.assertEquals(homePage.getHeader().getActualHeaderNavBarItems().size(),
+            EXPECTED_HEADER_NAV_BAR_ITEM_TEXTS.size());
+        softAssert.assertTrue(homePage.getHeader().getActualHeaderNavBarItemsTexts(homePage
+                                          .getHeader().getActualHeaderNavBarItems())
+                                      .containsAll(EXPECTED_HEADER_NAV_BAR_ITEM_TEXTS));
```

Comment:
Move all long chaining for such elements into additional variable. It looks to complicated now.